TemplePlus/protos.cpp

```diff
@@ -431,3 +435,54 @@ int ProtosHooks::ParseSpell(int colIdx, objHndl handle, char* content, obj_f fie
 
 	return 1;
 }
+
+int ProtosHooks::SetCritterAttacks(objHndl handle)
+{
+	auto obj = objSystem->GetObject(handle);
+	int32_t objType = obj->GetInt32(obj_f_type);
+	int result = 0;
+	if (objType == 13 || (result = objType, result == 14))
+	{
+		int32_t strAbility = obj->GetInt32(obj_f_critter_abilities_idx, 0); //v2 = getArrayFieldInt32(obj, obj_f_critter_abilities_idx, 0);
+		int strMod = objects.GetModFromStatLevel(strAbility);
+		int32_t sizeCategory = obj->GetInt32(obj_f_size);
+		int sizeMod = critterSys.GetBonusFromSizeCategory(sizeCategory);
+		int attackIndex = 0;
+		do
+		{
+			if (obj->GetInt32(obj_f_critter_attacks_idx, attackIndex) > 0)
+			{
+				int32_t attackBonusOld = obj->GetInt32(obj_f_attack_bonus_idx, attackIndex);
+				int32_t v6 = obj->GetInt32(obj_f_critter_damage_idx, attackIndex);
+				auto dice = Dice::FromPacked(v6);
+				auto diceCount = dice.GetCount(); //v15 = GetPackedDiceNumDice(v6);
+				auto diceSides = dice.GetSides(); //v14 = GetPackedDiceType(v6);
+				auto diceMod = dice.GetModifier(); //v7 = GetPackedDiceBonus(v6);
+				auto attackBonusNew = attackBonusOld - (strMod + sizeMod);
+				Dice dice2;
+				if (attackIndex <= 0 || strMod <= 0)
+					dice2 = Dice(diceCount, diceSides, diceMod - strMod); //v8 = encodeTriplet(v15, v14, v7 - v17);
+				else
+					dice2 = Dice(diceCount, diceSides, diceMod - strMod / 2); //v8 = encodeTriplet(v15, v14, v7 - v17 / 2);
+				auto dicePacked = dice2.ToPacked();
+				obj->SetInt32(obj_f_attack_bonus_idx, attackIndex, attackBonusNew);
+				obj->SetInt32(obj_f_critter_damage_idx, attackIndex, dicePacked);
+			}
+			++attackIndex;
+		} while (attackIndex < 3);
+
+		auto dexAbility = obj->GetInt32(obj_f_critter_abilities_idx, 1);
+		result = objects.GetModFromStatLevel(dexAbility);
+		for (int i = result; attackIndex < 4; ++attackIndex)
```

Comment:
This too

Reply:
Set Critter Attacks is handy function to remove str from damage dice and from attack bonus, because it will later be added during actual attack.

But the last 4th natural attach seems to be reserved specifically for Weapon Finesse attacks. It only removes dex mod from attack bonus. But it does not remove str mod from damage dice. Which is odd, as it still is going to affect damage dice.

**I suggest add str damage bonus removal similarly to 3 first natural attacks.** 

Arguments:
1) 4th attack is almost never used in protos. Exception - one dragon and one dude.
2) 4th attack even if it was specified in the protos was actually never used in vanilla (thus one of fixes above);
3)  Let's say I have Small Centipede with 1d4-3 damage. Without this fix I will need to remember that in 4 attack I need to manually change -3 to 0 (to simply 1d4), otherwise in reality during damage it will be 1d4-6 (as str mod is -3).